docs/authentication/social-connections/custom.mdx

```diff
@@ -0,0 +1,71 @@
+---
+title: Custom
+description: Configure a custom OpenID Connect (OIDC) compatible authentication provider for your Clerk application.
+---
+
+# Custom Auth Provider
+
+<Callout type="warning">
+  Custom auth providers are in beta. Please contact Support to get your application whitelisted for this feature.
+</Callout>
+
+Learn how to set up a custom authentication provider with Clerk.
+
+## Overview
+
+First, make sure your identity provider is compatible with OpenID Connect (OIDC), as Clerk relies on this protocol to authenticate users and retrieve their profile information.
+
+## Prerequisites
+
+- A Clerk Application created in your [Clerk Dashboard](https://dashboard.clerk.com/). For setup instructions, see our [Set up your application guide](/docs/quickstarts/setup-clerk).
+- Access to an OIDC-compliant identity provider.
+
+## Configuration Steps
+
+1. In the Clerk Dashboard, navigate to **User & Authentication > [Social Connections](https://dashboard.clerk.com/last-active?path=user-authentication/social-connections)**.
+1. Under **Custom Auth Providers**, click **Add provider**.
+1. Fill in the following fields in the modal:
+
+- **Name**: The provider name (visible to users)
+- **Key**: A unique identifier for the provider (cannot be changed after creation)
+- **Discovery Endpoint**: The OIDC discovery endpoint URL of your provider
+- Alternatively, click **Manual configuration** to set up provider endpoints manually
+- **Client ID**: Obtained from your provider
+- **Client Secret**: Obtained from your provider
+
+4. Click **Save** to add the provider to your instance.
+
+\[Screenshot of the Add provider modal]
+
+The provider is now configured but not yet enabled. Access the details page for advanced configuration options and to find the **Authorized redirect URLs** to configure in your provider's settings.
+
+Enable the provider either from the provider list or the top of the details page when ready.
+
+## Attribute Mapping
+
+If your provider returns claims in a non-standard format:
+
+1. Go to the provider's details page.
+1. Under **Attribute mapping**, configure the mapping to match your provider's claim format.
+
+### Tips
+
+1. For fields like **Email address verified**, set default values for missing claims.
+1. If the provider returns a claim but you don't want to set it, leave the mapping field empty.
+1. If you set a user info URL, it takes priority over the ID Token for claim retrieval.
+
+\[Screenshot of the Attribute mapping section]
```

Comment:
these aren't steps that occur in order, so we don't need numbered steps. we can use bullet points!
and remove the screenshot

```suggestion
- For fields like **Email address verified**, set default values for missing claims.
- If the provider returns a claim but you don't want to set it, leave the mapping field empty.
- If you set a user info URL, it takes priority over the ID Token for claim retrieval.
```

Reply:
Ah that's great 👌🏼